poc/vdaf_pine.py

```diff
@@ -415,135 +425,125 @@ def prep_next(Pine, prep_state, prep_msg):
-                          meas_len: Unsigned) -> list[Flp.Field]:
+                          meas_len: int) -> list[F]:
         """
         Generate the helper measurement share up to length `meas_len`,
         for aggregator ID `agg_id`, with measurement share seed `k_share`.
         """
-        return Pine.Flp.Valid.Xof.expand_into_vec(
-            Pine.Flp.Field,
+        return self.xof.expand_into_vec(
+            self.flp.field,
             k_share,
-            Pine.domain_separation_tag(USAGE_MEAS_SHARE),
+            self.domain_separation_tag(USAGE_MEAS_SHARE),
             byte(agg_id),
             meas_len,
         )
 
-    @classmethod
-    def helper_proofs_share(Pine,
-                            agg_id: Unsigned,
-                            k_share: bytes) -> list[Flp.Field]:
+    def helper_proofs_share(self,
+                            agg_id: int,
+                            k_share: bytes) -> list[F]:
         """
         Generate the helper proofs share for aggregator ID `agg_id`, with
         proof share seed `k_share`.
         """
-        return Pine.Flp.Valid.Xof.expand_into_vec(
-            Pine.Flp.Field,
+        return self.xof.expand_into_vec(
+            self.flp.field,
             k_share,
-            Pine.domain_separation_tag(USAGE_PROOF_SHARE),
-            byte(Pine.PROOFS) + byte(agg_id),
-            Pine.Flp.PROOF_LEN * Pine.PROOFS
+            self.domain_separation_tag(USAGE_PROOF_SHARE),
+            byte(self.PROOFS) + byte(agg_id),
+            self.flp.PROOF_LEN * self.PROOFS
         )
 
-    @classmethod
     def expand_input_share(
-        Pine,
-        agg_id: Unsigned,
-        input_share: InputShare,
-    ) -> tuple[list[Flp.Field], list[Flp.Field], bytes, bytes]:
+        self,
+        agg_id: int,
+        input_share: PineInputShare[F],
+    ) -> tuple[list[F], list[F], bytes, bytes]:
         """Expand Helper's seed into a vector of field elements. """
-        (
-            meas_share,
-            proofs_share,
-            k_wr_joint_rand_blind,
-            k_vf_joint_rand_blind
-        ) = input_share
         if agg_id > 0:
-            meas_share = \
-                Pine.helper_meas_share(agg_id, meas_share, Pine.MEAS_LEN)
-            proofs_share = Pine.helper_proofs_share(agg_id, proofs_share)
-        return (meas_share,
-                proofs_share,
+            (
+                k_meas_share,
+                k_proofs_share,
                 k_wr_joint_rand_blind,
-                k_vf_joint_rand_blind)
+                k_vf_joint_rand_blind,
+            ) = input_share
+            return (
+                self.helper_meas_share(agg_id, cast(bytes, k_meas_share), self.MEAS_LEN),
+                self.helper_proofs_share(agg_id, cast(bytes, k_proofs_share)),
+                k_wr_joint_rand_blind,
+                k_vf_joint_rand_blind,
+            )
+        return cast(tuple[list[F], list[F], bytes, bytes], input_share)
```

Comment:
Reviewer note: This is a situation where the code was taking advantage of Python's dynamic types. When we start enforcing type hints, the type of a variable is not supposed to change while it's in scope.